lib/propolis-client/src/support.rs

```diff
@@ -572,4 +584,21 @@ mod test {
     {
         WebSocketStream::from_raw_socket(conn, Role::Server, None).await
     }
+
+    #[test]
+    fn test_nvme_serial_from_str() {
+        use super::nvme_serial_from_str;
+
+        let expected = b"hello world         ";
+        assert_eq!(nvme_serial_from_str("hello world", b' '), *expected);
+
+        let expected = b"enthusiasm!!!!!!!!!!";
+        assert_eq!(nvme_serial_from_str("enthusiasm", b'!'), *expected);
```

Comment:
it is very funny to me that you have a test for this but _not_ `nvme_serial_from_str("these_are_not_made", 0x00)`. it'll definitely work for null-padding as-is, but that's the one non-space case we _do_ expect :D

Reply:
Yeah, we definitely want to cover that one! Fixed in d6dd7c7.